app/helpers/sites_helper.rb

```diff
@@ -1,5 +1,24 @@
 module SitesHelper
 
+  # Generate a text field control for form "site_form" for submitting the value
+  # of attribute "attribute" of the Site model being edited.  Use options passed
+  # in "common_options", but disable the control if the Site "@site" is not a
+  # point.
+  def geometry_aware_text_field(site_form, attribute, common_options)
+
+    if @site.point?
+      options = common_options
+    else
+      # Use :dummy (which returns "") as the attribute to force display of the placeholder.
+      attribute = :dummy
+      options = common_options.merge(disabled: true,
+                                     placeholder: "Site coordinates can't be updated for non-point sites.")
```

Comment:
Only suggestion would be to put the centroid values in, and have a note at the top that says something more like "non-point site; use SQL to update or contact administrator for more information"